python-frontend/src/main/java/org/sonar/python/tree/PythonTreeMaker.java

```diff
@@ -1299,14 +1295,17 @@ private Expression stringLiteral(AstNode astNode) {
     List<Expression> res = new ArrayList<>();
     parser.setRootRule(parser.getGrammar().rule(PythonGrammar.EXPR));
     // get escaped interpolation
-    Matcher matcher = INTERPOLATED_EXPR_PATTERN.matcher(literalValue);
-    while (matcher.find()) {
-      AstNode parse = parser.parse(matcher.group(2));
+    Matcher matcher = INTERPOLATED_EXPR_START_PATTERN.matcher(literalValue);
+    int index = 0;
+    while (matcher.find(index)) {
+      int end = matcher.end();
+      AstNode parse = parser.parse(literalValue.substring(end));
```

Comment:
Isn't it parsing all the remaining string literal here?
Like, for `f"{x} is equal to {y}"` it will parse `x} is equal to {y}` instead of just `x`

Reply:
indeed input of `parser.parse` will be `x} is equal to {y}` but this parser will eventually return tree corresponding to `x` only as the rest does not correspond to the grammar